File: hash-table/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        //I will use voting algo instead for this to get O(n)

        int count = 0;
        int candidate = 0;

        for (int i = 0; i < nums.length; i++){
            if(count == 0){
                candidate = nums[i];
            }

            if(candidate == nums[i]){
                count++;
            }else{
                count--;
            }
        }
        return candidate; 
    }
}

File: hash-table/majority-element.py
class Solution:
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        #hashmap 
        freq = {}
        max_freq = float('-inf')
        max_num = 0
        
        for num in nums:
            freq[num] = freq.get(num, 0) + 1
            
        #return the highest freq
        for num, value in freq.items():
            if value > max_freq:
                max_freq = value
                max_num = num
                
        return max_num
